libraries/src/test/java/com/baeldung/flink/WordCountTest.java

```diff
@@ -0,0 +1,120 @@
+package com.baeldung.flink;
+
+import org.apache.flink.api.common.functions.ReduceFunction;
+import org.apache.flink.api.java.DataSet;
+import org.apache.flink.api.java.ExecutionEnvironment;
+import org.apache.flink.api.java.functions.KeySelector;
+import org.apache.flink.api.java.operators.DataSource;
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.flink.api.java.tuple.Tuple3;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+
+public class WordCountTest {
+    final ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+
+    @Test
+    public void givenDataSet_whenExecuteWordCount_thenReturnWordCount() throws Exception {
+        //given
+        List<String> lines = Arrays.asList("This is a first sentence", "This is a second sentence with a one word");
+
+        //when
+        DataSet<Tuple2<String, Integer>> result = WordCount.startWordCount(env, lines);
+
+        //then
+        List<Tuple2<String, Integer>> collect = result.collect();
+        assertThat(collect.size()).isEqualTo(9);
+        assertThat(collect.contains(new Tuple2<>("a", 3))).isTrue();
+        assertThat(collect.contains(new Tuple2<>("sentence", 2))).isTrue();
+        assertThat(collect.contains(new Tuple2<>("word", 1))).isTrue();
+        assertThat(collect.contains(new Tuple2<>("is", 2))).isTrue();
+        assertThat(collect.contains(new Tuple2<>("this", 2))).isTrue();
+        assertThat(collect.contains(new Tuple2<>("second", 1))).isTrue();
+        assertThat(collect.contains(new Tuple2<>("first", 1))).isTrue();
+    }
+
+    @Test
+    public void givenListOfAmounts_whenUseMapReduce_thenSumAmountsThatAreOnlyAboveThreshold() throws Exception {
+        //given
+        DataSource<Integer> amounts = env.fromElements(1, 29, 40, 50);
```

Comment:
when you create a source, then you get DataSource, when you apply transformations on it, then you get DataSet, so this is correct